test/e2e/metrics_grabber_test.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"strings"
+
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/metrics"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// Missing = Assumed minus Observed, Invalid = Observed minus Assumed
+func validateLabelSet(labelSet map[string][]string, data metrics.Metrics, missing sets.String, invalidLabels map[string]sets.String, missingLabels map[string]sets.String) {
+	for metric, labels := range labelSet {
+		unobservedLabels := sets.NewString()
+		for _, label := range labels {
+			unobservedLabels.Insert(label)
+		}
+		vector, found := data[metric]
+		Expect(found).To(Equal(true))
+		if len(vector) == 0 {
```

Comment:
Failed Expect will just log an error, not stop the function - it will still be running